File: 169-majority-element/majority-element.c
int majorityElement(int* nums, int numsSize) {
    int candidate = 0; 
    int votecount = 0;

    for(int i = 0;i<numsSize;i++){
        if(votecount == 0){
            candidate = nums[i]; // assigning the value to the candidate
        }
        if(nums[i] == candidate){ // checking repeating candidate
            votecount++; // if find then add the count
        }else{
            votecount--;// if not then cancel the count with new value
        }
    }
    return candidate;
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        candidate = 0
        votecount = 0

        for i in range(n):
            if votecount == 0:
                candidate = nums[i]

            if nums[i] == candidate:
                votecount += 1
            else:
                votecount -= 1

        return candidate
        